### PR TITLE
Add basic webhook support, refactor rest of library to accomadate

### LINKED ISSUE #4: Webhooks Support?
Now that we have stable API support, we have the oppertunity to focus on webhooks as well. Slacks webhooks were built to be simple to use, and adding support to this library would be trivial.

Is this something that people want?
Should support exist within the API client itself, or as a seperate client?

```js
var slackAPIClient = SlackAPI.getAPIClient('access_token');
var slackWebhookClient = SlackAPI.getWebhookClient('http://webhook.url');
slackAPIClient.send('postMessage', // ...
slackWebhookClient.send({/* data */});
// or...
var slackClient = SlackAPI.getClient('access_token');
slackClient.api('postMessage', // ...
slackClient.webhook('http://webhook.url', { // ...
```

While it would be nice to just have a single client instance for both webhooks and API, the complexity of working with both access_tokens and webhook URLs at once makes me think that a separate design would be better.

Thoughts?

### PR DESCRIPTION
**Pull request by FredKSchott**
This new interface will most likely become the v1.0.0 release, since the rest of the library appears stable and has spent a few weeks being used in production at Worklife.

Resolves #4 "Webhooks Support"

- [NEW] getWebhookClient()
- apiClient.api() -> apiClient.send()
- getClient() -> getAPIClient()
- Updated in-code documentation
- Updated README

### CODE REVIEW

File: lib/api-client.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ SlackAPIClient.prototype.SlackError = SlackError;
  * @param {Function} [callback] An optional callback to propagate the response to.
  * @callback {[Error, ResponseBody, Response]}
  */
-SlackAPIClient.prototype.api = function(method, options, callback) {
```
Comment (jkresner) [inline lib/api-client.js]: Yeah this property was ugly. Maybe .call ?

Comment (FredKSchott) [inline lib/api-client.js]: Agreed, do you like `.call()` more than `.send()`? Whatever it is, I want it to be the same for both API & Webhook clients

Comment (FredKSchott) [inline lib/api-client.js]: Ah wait no, I forgot call is already taken by JavaScript https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call

.send() is clear enough.

Comment (jkresner) [inline lib/api-client.js]: Will give more feedback / ideas over the next week. Roll with it :)

ᐧ

On Mon, Jun 15, 2015 at 10:16 AM, Fred K. Schott <notifications@github.com>
wrote:

> In lib/api-client.js
> <https://github.com/FredKSchott/slackey/pull/9#discussion_r32443528>:
>
> > @@ -32,7 +32,7 @@ SlackAPIClient.prototype.SlackError = SlackError;
> >   * @param {Function} [callback] An optional callback to propagate the response to.
> >   * @callback {[Error, ResponseBody, Response]}
> >   */
> > -SlackAPIClient.prototype.api = function(method, options, callback) {
>
> Ah wait no, I forgot call is already taken by JavaScript
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call
>
> .send() is clear enough.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FredKSchott/slackey/pull/9/files#r32443528>.
>



-- 
*Jonathon Kresner*
*Co-founder, CEO*
@AirPair <http://twitter.com/airpair> | *airpair.com
<http://www.airpair.com>*
@hackerpreneur <http://twitter.com/hackerpreneur>

**Comment by FredKSchott** [comment]
Thanks for the feedback @jkresner!

**Pull request merged**